File: BE-FindingJob/src/main/java/com/example/befindingjob/entity/Job.java
package com.example.befindingjob.entity;

import com.example.befindingjob.entity.enumm.JobType;
import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "jobs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "jobId")
public class Job {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer jobId;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String requirements;

    private String salary;
    private String location;

    @Enumerated(EnumType.STRING)
    private JobType jobType;

    private String experienceRequired;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private java.time.LocalDateTime postingDate;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private java.time.LocalDateTime expirationDate;

    @JsonIgnore
    @ManyToMany
    @JoinTable(
            name = "job_skills",
            joinColumns = @JoinColumn(name = "job_id"),
            inverseJoinColumns = @JoinColumn(name = "skill_id")
    )
    private Set<Skill> requiredSkills = new HashSet<>();

    @ManyToOne
    @JsonBackReference
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @JsonIgnore
    @OneToMany(mappedBy = "job", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Application> applications = new HashSet<>();

    @JsonIgnore
    @ManyToMany(mappedBy = "bookmarkedJobs")
    private Set<User> usersBookmarked = new HashSet<>();
}

File: JobHub/app/src/main/java/com/example/jobhub/entity/Job.kt
package com.example.jobhub.entity

import android.os.Parcelable
import com.example.jobhub.entity.enumm.JobType
import kotlinx.parcelize.Parcelize

@Parcelize
data class Job(
    var jobId: Int = 0,
    var title: String? = null,
    var description: String? = null,
    var requirements: String? = null,
    var salary: String? = null,
    var location: String? = null,
    var jobType: JobType? = null,
    var experienceRequired: String? = null,
    var postingDate: String? = null,
    var expirationDate: String? = null,
    var requiredSkills: Set<Skill>? = null,
    var company: Company? = null,
    var applications: Set<Application>? = null
) : Parcelable
